--- a/arch/arm/boot/dts/rockchip/rv1126.dtsi
+++ b/arch/arm/boot/dts/rockchip/rv1126.dtsi
@@ -68,6 +68,20 @@
 		};
 	};
 
+	wdt: watchdog@ff680000 {
+		compatible = "rockchip,rv1126-wdt", "snps,dw-wdt";
+		reg = <0xff680000 0x100>;
+		interrupts = <GIC_SPI 32 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cru PCLK_WDT>;
+		resets = <&cru SRST_WDT_P>;
+		/* Optional: only if you want to provide explicit TOPs */
+		snps,watchdog-tops = <0x0001000 0x0002000 0x0004000 0x0008000
+					0x0010000 0x0020000 0x0040000 0x0080000
+					0x0100000 0x0200000 0x0400000 0x0800000
+					0x1000000 0x2000000 0x4000000 0x8000000>;
+		status = "disabled";
+	};
+	
 	arm-pmu {
 		compatible = "arm,cortex-a7-pmu";
 		interrupts = <GIC_SPI 123 IRQ_TYPE_LEVEL_HIGH>,
--- a/arch/arm/boot/dts/rockchip/rv1126-sonoff-ihost.dts
+++ b/arch/arm/boot/dts/rockchip/rv1126-sonoff-ihost.dts
@@ -27,3 +27,7 @@
 &cpu3 {
 	cpu-supply = <&vdd_arm>;
 };
+
+&wdt {
+	status = "okay";
+};
\ No newline at end of file
